JS/02/table/table.js

```diff
@@ -0,0 +1,215 @@
+  if(confirm('Bạn có muốn xóa user ?')) {
+    users.splice(index,1)
+  }
+  render(users)
+}
+
+const editUser = (e, index) => {
+  const activeEdit = e.target
+  const patentEdit = activeEdit.parentElement
+  const activeTableinSave = patentEdit.querySelector('.box_active-table--save')
+  const activeTableinCancel = patentEdit.querySelector('.box_active-table--cancel')
+  activeEdit.classList.add('d-none')
+  activeTableinSave.classList.add('d-inline-block')
+  activeTableinCancel.classList.add('d-inline-block')
+
+  const boxParent = getParent(activeTableinSave, 'tr')
+  const fixName = boxParent.querySelector('td:nth-child(3)')
+  const fixGender = boxParent.querySelector('td:nth-child(4)')
+  const fixAge = boxParent.querySelector('td:nth-child(5)')
+
+  users.forEach((user, i) => {
+    if (index === i) {
+      fixName.innerHTML = `
+        <input type="text"
+          name="" id="input"
+          class="form-control full-name"
+          value="${user.fullName}">`
+      if (user.gender === 'male') {
+        fixGender.innerHTML = `
+          <select class="form-control gender" name="" id="">
+            <option value="male">Male</option>
+            <option value="female">Female</option>
+          </select>`
+      } else {
+        fixGender.innerHTML = `
+          <select class="form-control gender" name="" id="">
+            <option value="female">Female</option>
+            <option value="male">Male</option>
+          </select>`
+      }
+      fixAge.innerHTML = `
+        <input type="number"
+          name=""
+          id="input"
+          class="form-control age"
+          value="${user.age}">`
+    }
+  })
+}
+
+// Tìm parent bên ngoài
+const getParent = (ele, parent) => {
+  while(ele.parentElement) {
+    if (ele.parentElement.matches(parent)) {
+      return ele.parentElement
+    }
+    ele = ele.parentElement
+  }
+}
+
+const editSave = (e, index) => {
+  const activeTableinSave = e.target
+  const boxParent = getParent(activeTableinSave, 'tr')
+  const genderNew = boxParent.querySelector('.gender').value
+  const nameNew = boxParent.querySelector('.full-name').value
+  const ageNew = boxParent.querySelector('.age').value
+
+  users.map((user, i) => {
+    if (index === i) {
+      user.fullName = nameNew
+      user.gender = genderNew
+      user.age = ageNew
+    }
+  })
+  render(users)
+}
+
+const editCancel = e => {
+  render(users)
+}
+
+const selectAll = () => {
+  const tableHeading = document.querySelector('.heading-table')
+  const checkAll = tableHeading.querySelector('input[type="checkbox"]')
+  const listUser = document.querySelector('.list-user')
+  const boxRow = listUser.querySelectorAll('tr')
+  boxRow.forEach(row => {
+    const select = row.querySelector('input[type="checkbox"]')
+    if (checkAll.checked) {
+      select.setAttribute('checked','checked')
+    } else {
+      select.removeAttribute('checked','checked')
+    }
+  })
+}
+
+const selectDelete = () => {
```

Comment:
function name is confused, should not contain 2 verbs
```suggestion
const deleteMany= () => {
```